Add and fix tests and pom.xml

agoevnauruz opened the pull request.

agoevnauruz committed:
Add and fix tests and pom.xml

pavel-ulan commented on src/main/java/org/example/client/OrderClient.java:
```diff
@@ -0,0 +1,62 @@
+                    .body(order)
+                    .when()
+                    .request(method, PATH)
+                    .then();
```
**нужно исправить**
не совсем правильно делать это через if else. особенно с учетом, что это метод-шаг. Разнеси по разным методам

pavel-ulan commented on src/main/java/org/example/client/OrderClient.java:
```diff
@@ -0,0 +1,62 @@
+
+    public OrderClient() {
+        ValidatableResponse responseLogin = userClient.login(CredentialProvider.getDefault());
+        accessToken = responseLogin.extract().path("accessToken").toString().substring(6).trim();
```
**можно лучше**
все же не стоит выносить создание и логин пользователя в конструктор. Даже в этом задании не для всех тестов нужен пользователь.

pavel-ulan commented on src/test/java/org/example/createOrderTest/CreateOrderTest.java:
```diff
@@ -0,0 +1,79 @@
+    @After
+    public void tearDown() throws InterruptedException {
+        userClient.delete(accessToken);
+        Thread.sleep(300);
```
**надо исправить**
Первое - использование sleep() довольно плохая практика.
второе - использование его после всех действий еще и бесполезная =)

pavel-ulan commented on src/test/java/org/example/createOrderTest/CreateOrderTest.java:
```diff
@@ -0,0 +1,79 @@
+        ValidatableResponse responseLogin = userClient.login(CredentialProvider.getDefault());
+        String accessToken = responseLogin.extract().path("accessToken");
+        ValidatableResponse response = userClient.delete(accessToken);
+        response.statusCode(200);
```
**можно лучше**
В принципе валидно и работать будет, но все же смотрится странно. Если уж придерживаешься использования Asserts, то и следуй этому. А если хочешь использовать проверки Rest-assured'а, то и в остальной работе так и делай. В общем - используй один подход в работе, иначе это будет только путать.

pavel-ulan commented on src/test/java/org/example/userCreateTests/UserCreateTest.java:
```diff
@@ -0,0 +1,34 @@
+    @After
+    public void tearDown() throws InterruptedException {
+        Thread.sleep(300);
+    }
```
**надо исправить**
у тебя этот метод встречается в нескольких тестах пустым как тут, для чего ты его оставляешь?

pavel-ulan reviewed (changes requested):
Неплохо, но есть ряд странных решений. Посмотри комменты и поправь пожалуйста.

agoevnauruz committed:
added some change after review:

- deleted useless methods
- change if-else to 2 different step-method
- clear code and delete empty imports

agoevnauruz committed:
added allure-report

agoevnauruz committed:
delete all useless sleep methods
